src/ucp/rma/flush.c

```diff
@@ -305,6 +305,13 @@ ucs_status_ptr_t ucp_ep_flush_internal(ucp_ep_h ep, unsigned uct_flags,
         return NULL;
     }
 
+    if (uct_flags & UCT_FLUSH_FLAG_CANCEL) {
+        /* flush cancel moves UCT EP to failed state.
+         * To avoid race between error flow and explicit cancelation
+         * mark UCP EP failed */
+        ep->flags |= UCP_EP_FLAG_FAILED;
```

Comment:
I do not think so, AFAIU it's needed to avoid scheduling `ucp_worker_iface_err_handle_progress` from `ucp_worker_set_ep_failed` when transport error occurs after flush(cancel) which is currently used only for close(force), where we should wait completions from transport and destroy all lanes similar to err_handling flow which does similar thing by discart